perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2201,22 +2201,41 @@ public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String
 		for (Attribute a : logins) {
 			if (a.getFriendlyNameParameter().equals(namespace)) found = true;
 		}
-		if (!found) throw new InternalErrorException(user.toString()+" doesn't have login in namespace: "+namespace);
+		if (!found)
+			throw new InternalErrorException(user.toString() + " doesn't have login in namespace: " + namespace);
 
 		String email = "";
 		try {
-			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF+":preferredMail");
+			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, mailAttributeUrn);
 			if (a != null && a.getValue() != null) {
-				email = (String)a.getValue();
+				email = (String) a.getValue();
 			}
 		} catch (WrongAttributeAssignmentException ex) {
 			throw new InternalErrorException(ex);
 		} catch (AttributeNotExistsException ex) {
 			throw new InternalErrorException(ex);
 		}
 
+		String subject;
+		try {
+			Attribute messageTemplateAttribute = perunBl.getAttributesManager().getAttribute(sess, language,
+					"urn:perun:entityless:attribute-def:def:nonAuthzPwdResetMailSubject:" + namespace);
+			subject = messageTemplateAttribute.getValue().toString();
+		} catch (PrivilegeException | AttributeNotExistsException | WrongAttributeAssignmentException ex) {
+			throw new InternalErrorException(ex);
+		}
+
+		String message;
+		try {
+			Attribute subjectTemplateAttribute = perunBl.getAttributesManager().getAttribute(sess, language,
+					"urn:perun:entityless:attribute-def:def:nonAuthzPwdResetMailTemplate:" + namespace);
+			message = subjectTemplateAttribute.getValue().toString();
+		} catch (PrivilegeException | AttributeNotExistsException | WrongAttributeAssignmentException ex) {
+			throw new InternalErrorException(ex);
+		}
+
```

Comment:
Please add fallback for both message and subject. When filling with passed namespace (eg. "cs") fails (attribute or its value is null), then you retry this with generic "en" value for the namespace. So we have configurable fallback in attribute with "en" namespace. Right now following implementation in Utils fallback to hardcoded text only when passed value is null.